Sources/Extensions/UIKit/UIButtonExtensions.swift

```diff
@@ -161,6 +161,16 @@ public extension UIButton {
 	public func setTitleForAllStates(_ title: String) {
 		states.forEach { self.setTitle(title, for: $0) }
 	}
+    
+    /// SwifterSwift: Center align title text and image on UIButton
+    ///
+    /// - Parameter spacing: spacing between UIButton title text and UIButton Image.
+    public func centerTextAndImage(spacing: CGFloat) {
+        let insetAmount = spacing / 2
+        imageEdgeInsets = UIEdgeInsets(top: 0, left: -insetAmount, bottom: 0, right: insetAmount)
+        titleEdgeInsets = UIEdgeInsets(top: 0, left: insetAmount, bottom: 0, right: -insetAmount)
```

Comment:
There's some reason for the title insets be the inverse of image insets(image negative on left and title negative on right )? I think it makes the image and goes in different directions increasing the spacing between them.

Reply:
yeah! that is purpose of this

Reply:
Nice :) Now that you say this is the purpose I'm thinking about the naming of this extension, what you think about `textAndImageCenterSpacing(_ spacing: )`? I think it makes the purpose more clear. Let me know what you think :)

Reply:
When I added this extension, I had two things in mind:
1. By default, if we set an image to button it aligns left but for nice UI we want to Center it just before the title text of the button. So the first purpose of this extension was **Center align Image and Title text on the button**.
2. We want some spacing between title text and image. so second purpose was **Add some spacing between center aligned image and title text**.

So can we name this `centerTextAndImage(with spacing: CGFloat)`?
@LucianoPAlmeida your views?